### PR TITLE
Assign envvars to Polarion report arguments

### PR DESCRIPTION
**Pull request by KwisatzHaderach**
For easier usage of tmt report polarion and being able to set these arguments also from TF I'm assigning environment variables to most of the arguments.
Adding option `--use-facts` to get hostname and architecture from guest facts.
Use consistent naming for compose id.

### CODE REVIEW

**Comment by KwisatzHaderach** [comment]
Probably best to just use the features introduced in #1579 which would use different names for the variables, e.g. TMT_PLUGIN_REPORT_POLARION_LOGS.

File: tmt/steps/report/polarion.py [hunk 1]
```diff
@@ -151,8 +192,15 @@ def go(self) -> None:
             'polarion-project-span-ids': project_id}
         for tr_field in other_testrun_fields:
             param = self.get(tr_field)
```
Comment (thrix) [inline tmt/steps/report/polarion.py:194]: Actually, below could set the default to the getenv? sparing few lines?

```
param = self.get(tr_field, os.getenv(f'TMT_PLUGIN_REPORT_POLARION_{tr_field.upper()}'))
```

File: tmt/steps/report/polarion.py [hunk 2]
```diff
@@ -42,70 +43,107 @@ class ReportPolarionData(tmt.steps.report.ReportStepData):
...
         option='--title',
         metavar='TITLE',
-        help='Use specific test run title.'
+        envvar='TMT_PLUGIN_REPORT_POLARION_TITLE',
+        help=(
+            'Use specific test run title, '
+            'also uses environment variable TMT_PLUGIN_REPORT_POLARION_TITLE.')
+        )
+
+    use_facts: bool = field(
```
Comment (thrix) [inline tmt/steps/report/polarion.py:52]: seems this functionality is not reflected in description of the MR / commit?

File: tmt/steps/report/polarion.py [hunk 3]
```diff
@@ -35,77 +36,106 @@ class ReportPolarionData(tmt.steps.report.ReportStepData):
         default=None,
         option='--project-id',
         metavar='ID',
-        help='Use specific Polarion project ID.'
+        help=(
+            'Use specific Polarion project ID.'
+            'also uses environment variable TMT_PLUGIN_REPORT_POLARION_PROJECT_ID.')
```
Comment (happz) [inline tmt/steps/report/polarion.py:41]: Missing leading space in this string, plus a dot finishing the sentence in the previous string => `... project ID.also ...`

File: tmt/utils.py [hunk 4]
```diff
@@ -4661,6 +4661,7 @@ def field(
         choices: Union[None, Sequence[str], Callable[[], Sequence[str]]] = None,
         multiple: bool = False,
         metavar: Optional[str] = None,
+        envvar: Optional[str] = None,
```
Comment (happz) [inline tmt/utils.py:4664]: Is the change to `tmt/utils.py` still needed?

Comment (KwisatzHaderach) [reply tmt/utils.py:4664]: it shouldn't hurt and maybe someone will want to use it at some point?
for this patch it's not needed, might be better to remove

File: tmt/steps/report/polarion.py [hunk 5]
```diff
@@ -35,77 +36,106 @@ class ReportPolarionData(tmt.steps.report.ReportStepData):
...
         metavar='LOGLOCATION',
-        help='Location of the logs for this test run'
+        help=(
+            'Location of the logs for this test run, '
+            'also uses environment variable TMT_PLUGIN_REPORT_POLARION_LOGS.')
         )
 
     composeid: Optional[str] = field(
         default=None,
         option='--composeid',
```
Comment (psss) [inline tmt/steps/report/polarion.py:134]:
```suggestion
        option='--compose-id',
```

Is it too late to make the option consistent with others and use dash to separate words?

**Review by psss — APPROVED** [review]
Looks good, just one question regarding consistent option naming.

File: tmt/steps/report/polarion.py (hunk 5 above)
Comment (KwisatzHaderach) [reply tmt/steps/report/polarion.py:134]: yeah, the problem is on polarion side, it's one word there and I don't want to do some magic splitting/merging on tmt side for one variable

**Pull request merged**